Implement _mm_shuffle_epi8 with NEON instruction

marktwtn opened the pull request:
Add the corresponding unit testing.

jserv commented on sse2neon.h:
```diff
@@ -1000,33 +1000,13 @@ FORCE_INLINE __m128i _mm_shuffle_epi_3332(__m128i a)
-    uint8_t *b = (uint8_t *) &ib;  // input b
-    int32_t r[4];
+    int8x16_t t = vreinterpretq_s8_m128i(a);     // input a
+    uint8x16_t idx = vreinterpretq_u8_m128i(b);  // input b
```
Can you replace `t` and `idx` with something mnemonic?

marktwtn replied:
I just follow the input argument naming by searching the `vqtbl1q_s8` in the [Neon intrinsic page](https://developer.arm.com/architectures/instruction-sets/simd-isas/neon/intrinsics).

I think the `t` variable can be changed to `tbl`, or remains the same to follow the description in the Neon intrinsic page.

jserv replied:
`tbl` sounds better since it is a matter of table lookup operations.

marktwtn replied:
Done.

jserv commented:
Please append corresponding test case.

marktwtn commented:
I have added the corresponding unit testing and modified the implementation of `_mm_shuffle_epi8`.
The modification is based on the difference behaviour of `_mm_shuffle_epi8` and `vqtbl1q_s8`.

However, the unit testing is still not passing.
The pull request should not be merged until I fix the bug.

marktwtn commented:
The pull request can be merged without problem now.
The unit testing has been fixed.
I wrote the wrong unit testing since I forgot the difference of the signed/unsigned integer casting.
